modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImplTest.java

```diff
@@ -430,49 +444,88 @@ private void _testExtractIndexableAttributes(
 				LocaleUtil.US));
 	}
 
-	private void _testFormWithRepeatableField(String indexType) {
+	private void _testFormWithRepeatableField(
+		String indexType, boolean enableLegacyDDMIndexFields) {
```

Comment:
@icgsbr simplify the code by creating a new method instead of using `if/else` statement:

```
@Test
public void testFormWithRepeatableField() {
	_testFormWithRepeatableField("keyword");
	_testFormWithRepeatableField("text");
	_testFormWithRepeatableRichTextField();
}

private void _testFormWithRepeatableField(String indexType) {
	Document document = _createDocument();

	DDMForm ddmForm = DDMStructureTestUtil.getSampleDDMForm(
		_FIELD_NAME, "string", indexType, true,
		DDMFormFieldTypeConstants.TEXT,
		new Locale[] {LocaleUtil.getSiteDefault()},
		LocaleUtil.getSiteDefault());

	_ddmIndexer.addAttributes(
		document, _createDDMStructure(ddmForm),
		_createDDMFormValues(
			ddmForm,
			DDMFormValuesTestUtil.createDDMFormFieldValue(
				_FIELD_NAME,
				DDMFormValuesTestUtil.createLocalizedValue(
					"able", LocaleUtil.US)),
			DDMFormValuesTestUtil.createDDMFormFieldValue(
				_FIELD_NAME,
				DDMFormValuesTestUtil.createLocalizedValue(
					"baker", LocaleUtil.US))));

	indexType = StringUtil.upperCaseFirstLetter(indexType);

	FieldValuesAssert.assertFieldValues(
		_getSortableValues(
			Collections.singletonMap(
				"ddmFieldArray.ddmFieldValue" + indexType + "_en_US",
				"[able, baker]")),
		"ddmFieldArray.ddmFieldValue" + indexType, document,
		StringPool.BLANK);
}

private void _testFormWithRepeatableRichTextField() {
	_ddmIndexer = _createDDMIndexer(true);

	Document document = _createDocument();

	DDMForm ddmForm = DDMStructureTestUtil.getSampleDDMForm(
		_FIELD_NAME, "string", "text", true,
		DDMFormFieldTypeConstants.RICH_TEXT,
		new Locale[] {LocaleUtil.getSiteDefault()},
		LocaleUtil.getSiteDefault());

	DDMStructure ddmStructure = _createDDMStructure(ddmForm);

	_ddmIndexer.addAttributes(
		document, ddmStructure,
		_createDDMFormValues(
			ddmForm,
			DDMFormValuesTestUtil.createDDMFormFieldValue(
				_FIELD_NAME,
				DDMFormValuesTestUtil.createLocalizedValue(
					"<h1>test</h1>", LocaleUtil.US))));

	Assert.assertEquals(
		"test",
		document.get(
			StringBundler.concat(
				"ddm__text__", ddmStructure.getStructureId(), "__",
				_FIELD_NAME, "_en_US")));
}
```